### PR TITLE
Create CreateTrainingSetRunner

### PR DESCRIPTION
**Pull request by Sami1309**

### CODE REVIEW

**Review by sdreyer — DISMISSED** [review]
lgtm

File: ff/serving/runner/copy_test.go [hunk 1]
```diff
@@ -288,7 +288,7 @@ type ErrorChunkRunnerFactoryConfigs struct {
 	ErrorConfig Config
 }
 
-func serialize(config MaterializedChunkRunnerConfig) Config {
+func copySerialize(config MaterializedChunkRunnerConfig) Config {
 	serializedConfig, _ := json.Marshal(config)
```
Comment (simba-git) [inline ff/serving/runner/copy_test.go:292]: Don't drop the error

Comment (simba-git) [reply ff/serving/runner/copy_test.go:292]: rename to serializeMaterializeConfig

File: ff/serving/runner/training_set_runner.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+func (c *CreateTrainingSetRunnerConfig) Deserialize(config Config) error {
+	err := json.Unmarshal(config, c)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func CreateTrainingSetRunnerFactory(config Config) (Runner, error) {
```
Comment (simba-git) [inline ff/serving/runner/training_set_runner.go:52]: TrainingSetRunnerFactory

File: ff/serving/runner/training_set_runner_test.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+	if err != nil {
+		t.Fatalf("failed to create create training set runner: %v", err)
+	}
+	if err := watcher.Wait(); err == nil {
+		t.Fatalf("failed to report error creating training set")
+	}
+}
+
+func trainingSetSerialize(config CreateTrainingSetRunnerConfig) Config {
+	serializedConfig, _ := json.Marshal(config)
```
Comment (simba-git) [inline ff/serving/runner/training_set_runner_test.go:65]: Use the Serialize() method

File: ff/serving/runner/training_set_runner.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+		if err := m.Offline.CreateTrainingSet(m.Def); err != nil {
+			trainingSetWatcher.EndWatch(err)
+			return
+		}
+		trainingSetWatcher.EndWatch(nil)
+	}()
+	return trainingSetWatcher, nil
+}
+
+type CreateTrainingSetRunnerConfig struct {
```
Comment (simba-git) [inline ff/serving/runner/training_set_runner.go:30]: Rename TrainingSetRunnerConfig

File: ff/serving/runner/training_set_runner_test.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+type ErrorTrainingSetFactoryConfigs struct {
+	ErrorType   string
+	ErrorConfig Config
+}
+
+func TestTrainingSetRunnerFactoryErrorCoverage(t *testing.T) {
+	errorConfigs := []ErrorTrainingSetFactoryConfigs{
+		ErrorTrainingSetFactoryConfigs{
+			ErrorType:   "cannot deserialize config",
```
Comment (simba-git) [inline ff/serving/runner/training_set_runner_test.go:82]: You shouldn't test error strings, they can change over time

File: ff/serving/runner/training_set_runner_test.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+type ErrorTrainingSetFactoryConfigs struct {
+	ErrorType   string
+	ErrorConfig Config
+}
+
+func TestTrainingSetRunnerFactoryErrorCoverage(t *testing.T) {
+	errorConfigs := []ErrorTrainingSetFactoryConfigs{
+		ErrorTrainingSetFactoryConfigs{
```
Comment (simba-git) [inline ff/serving/runner/training_set_runner_test.go:81]: You can remove CreateTrainingSetRunnerConfig

**Review by simba-git — COMMENTED** [review]
Looks good, just some comments on naming

File: ff/serving/runner/training_set_runner_test.go (hunk 6 above)
Comment (Sami1309) [reply ff/serving/runner/training_set_runner_test.go]: changed

File: ff/serving/runner/training_set_runner_test.go (hunk 5 above)
Comment (Sami1309) [reply ff/serving/runner/training_set_runner_test.go]: changed to "Name"

File: ff/serving/runner/training_set_runner.go (hunk 4 above)
Comment (Sami1309) [reply ff/serving/runner/training_set_runner.go]: renamed

File: ff/serving/runner/training_set_runner_test.go (hunk 3 above)
Comment (Sami1309) [reply ff/serving/runner/training_set_runner_test.go]: changed to nested function that tests error and calls Serialize()

File: ff/serving/runner/training_set_runner.go (hunk 2 above)
Comment (Sami1309) [reply ff/serving/runner/training_set_runner.go]: changed

File: ff/serving/runner/copy_test.go (hunk 1 above)
Comment (Sami1309) [reply ff/serving/runner/copy_test.go]: function removed

**Pull request merged**